travis-ci/managers/debian.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+PHASES=(${@:-SETUP RUN RUN_ASAN CLEANUP})
+DEBIAN_RELEASE="${DEBIAN_RELEASE:-testing}"
+CONT_NAME="${CONT_NAME:-debian-$DEBIAN_RELEASE-$RANDOM}"
+DOCKER_EXEC="${DOCKER_EXEC:-docker exec -it $CONT_NAME}"
+DOCKER_RUN="${DOCKER_RUN:-docker run}"
+REPO_ROOT="${REPO_ROOT:-$PWD}"
+ADDITIONAL_DEPS=(clang)
+
+function info() {
+    echo -e "\033[33;1m$1\033[0m"
+}
+
+set -e
+
+source "$(dirname $0)/travis_wait.bash"
+
+for phase in "${PHASES[@]}"; do
+    case $phase in
+        SETUP)
+            info "Setup phase"
+            info "Using Debian $DEBIAN_RELEASE"
+	    docker pull debian:$DEBIAN_RELEASE
+            info "Starting container $CONT_NAME"
+            $DOCKER_RUN -v $REPO_ROOT:/build:rw \
+                        -w /build --privileged=true --name $CONT_NAME \
+			-dit --net=host debian:$DEBIAN_RELEASE /bin/bash
+            $DOCKER_EXEC bash -c "echo deb-src http://deb.debian.org/debian $DEBIAN_RELEASE main >>/etc/apt/sources.list"
+            $DOCKER_EXEC apt-get -y update
+            $DOCKER_EXEC apt-get -y build-dep libelf-dev
+            $DOCKER_EXEC apt-get -y install libelf-dev
+            $DOCKER_EXEC apt-get -y install "${ADDITIONAL_DEPS[@]}"
+            ;;
+        RUN|RUN_CLANG)
+            if [[ "$phase" = "RUN_CLANG" ]]; then
+                ENV_VARS="-e CC=clang -e CXX=clang++"
+            fi
+            docker exec $ENV_VARS -it $CONT_NAME mkdir build
+            $DOCKER_EXEC make CFLAGS='-Werror' -C ./src -B OBJDIR=../build
+            $DOCKER_EXEC rm -rf build
+            ;;
+        RUN_ASAN|RUN_CLANG_ASAN)
+            if [[ "$phase" = "RUN_CLANG_ASAN" ]]; then
+                ENV_VARS="-e CC=clang -e CXX=clang++"
+            fi
+            docker exec $ENV_VARS -it $CONT_NAME mkdir build
+            $DOCKER_EXEC make CFLAGS='-Werror -Db_sanitize=address,undefined' -C ./src -B OBJDIR=../build
```

Comment:
@wat-ze-hex thank you for bringing Travis CI to the libbpf repository! 

`-Db_sanitize` is [a `meson`-ism](https://mesonbuild.com/howtox.html#use-address-sanitizer). To build `libbpf` with `ASan+UBSan` using `make` it seems that something like `-fsanitize=address,undefined` should be used instead.

It also looks like that CFLAGS passed via the command line overrides whatever is set in `Makefile` (including `-Wall`), which I think makes `-Werror` a little bit less useful.

`ENV_VARS` don't seem to affect `make` in the sense that `make` runs `gcc` even when `phase` is `RUN_CLANG*`. It would probably be better to use something like `docker exec $ENV_VARS make CFLAGS='...' -C ./src -B OBJDIR=../build`.

And, lastly, to judge from the logs `make` complained that "make: pkg-config: Command not found" but I'm not sure how critical this is.

Reply:
@wat-ze-hex could you have a followup on this?

Reply:
@evverx great catch!

`Makefile` from libbpf works with env vars just fine, e.g.
```
14:43:46 1 rdna@:~/.../git/libbpf$>make -C src/ -n | head -n 3
make: Entering directory `/data/users/rdna/git/libbpf/src'
cc -I. -I../include -I../include/uapi -DCOMPAT_NEED_REALLOCARRAY -fPIC -fvisibility=hidden -g -O2 -Werror -Wall    -c bpf.c -o bpf.o
cc -I. -I../include -I../include/uapi -DCOMPAT_NEED_REALLOCARRAY -fPIC -fvisibility=hidden -g -O2 -Werror -Wall    -c btf.c -o btf.o
14:43:48 0 rdna@:~/.../git/libbpf$>env CC=~/bin/clang make -C src/ -n | head -n 3
make: Entering directory `/data/users/rdna/git/libbpf/src'
/home/rdna/bin/clang -I. -I../include -I../include/uapi -DCOMPAT_NEED_REALLOCARRAY -fPIC -fvisibility=hidden -g -O2 -Werror -Wall    -c bpf.c -o bpf.o
/home/rdna/bin/clang -I. -I../include -I../include/uapi -DCOMPAT_NEED_REALLOCARRAY -fPIC -fvisibility=hidden -g -O2 -Werror -Wall    -c btf.c -o btf.o
```
The problem here is that `$ENV_VARS` is passed  to `mkdir` on line 47, but it should be passed to actual `make` on line 48 instead. That should be fixed indeed.

As for `CFLAGS` it was made this way intentionally so that user can fully override flags that are not required for the build. See a82a66eda3b1965d786960b61c1899011a15368d for details. So yeah, `-Werror` and other necessary flags should be passed explicitly.

Reply:
@evverx

Thanks, created a follow-up PR!